### PR TITLE
Dataset custom query

### PR DESCRIPTION
**Pull request by StanislasGuinel**
[CH59793] Custom query parameter for Dataset component 

Wrap the user query in a CALL {USER_QUERY} RETURN * LIMIT XX for previewing

### CODE REVIEW

File: python-connectors/neo4j/connector.py [hunk 1]
```diff
@@ -66,9 +65,18 @@ def generate_rows(self, dataset_schema=None, dataset_partitioning=None, partitio
                     yield {
                         "source node": rel.start_node["name"],
                         "relationship": type(rel).__name__,
-                        "target node": rel.end_node["name"]
+                        "target node": rel.end_node["name"],
                     }
 
+        elif self.config["select_nodes_or_relationships"] == "custom_query":
+            custom_query = self.config.get("selected_custom_query", None)
```
Comment (alexbourret) [inline python-connectors/neo4j/connector.py:72]: Maybe you could raise and point to the doc if custom_query is None ?

File: python-connectors/neo4j/connector.py [hunk 2]
```diff
@@ -29,10 +31,7 @@ def generate_rows(self, dataset_schema=None, dataset_partitioning=None, partitio
                     query += " LIMIT {}".format(records_limit)
                 data = self.graph.run(query)
                 for row in data:
-                    node_properties = {
-                        "node_label": self.config["selected_node"],
-                        "neo4j_id": row["n"].identity
-                    }
+                    node_properties = {"node_label": self.config["selected_node"], "neo4j_id": row["n"].identity}
```
Comment (alexbourret) [inline python-connectors/neo4j/connector.py:34]: Maybe replacing all the "n", "r", "name"... with a NODE, RELATIONSHIP... would make it clearer while keeping the actual queries short.

File: python-connectors/neo4j/connector.py [hunk 3]
```diff
@@ -66,9 +65,18 @@ def generate_rows(self, dataset_schema=None, dataset_partitioning=None, partitio
...
+                        "target node": rel.end_node["name"],
                     }
 
+        elif self.config["select_nodes_or_relationships"] == "custom_query":
+            custom_query = self.config.get("selected_custom_query", None)
+            logging.info(f"Neo4j plugin - Dataset - User custom query: {custom_query}")
+            if records_limit > -1:
+                custom_query = f"CALL {{{custom_query}}} RETURN * LIMIT {records_limit}"
+            results = self.graph.run(custom_query)
+            for record in results.data():
```
Comment (alexbourret) [inline python-connectors/neo4j/connector.py:77]: Just to make sure: there is no pagination mechanism ?

Comment (StanislasGuinel) [reply python-connectors/neo4j/connector.py:77]: I will look for more efficient way to retrieve large data

Comment (StanislasGuinel) [reply python-connectors/neo4j/connector.py]: Using `for row in session.run(query)` to get the rows one by one from the neo4j server.

**Pull request merged**